OutOfSchool/OutOfSchool.IdentityServer/Services/MinistryAdminService.cs

```diff
@@ -0,0 +1,335 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using OutOfSchool.Common;
+using OutOfSchool.Common.Models;
+using OutOfSchool.EmailSender;
+using OutOfSchool.IdentityServer.Services.Intefaces;
+using OutOfSchool.IdentityServer.Services.Interfaces;
+using OutOfSchool.IdentityServer.Services.Password;
+using OutOfSchool.RazorTemplatesData.Models.Emails;
+using OutOfSchool.RazorTemplatesData.Services;
+using OutOfSchool.Services;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+
+namespace OutOfSchool.IdentityServer.Services
+{
+    public class MinistryAdminService : IMinistryAdminService
+    {
+        private readonly IEmailSender emailSender;
+        private readonly IMapper mapper;
+        private readonly ILogger<MinistryAdminService> logger;
+        private readonly IMinistryAdminRepository ministryAdminRepository;
+
+        private readonly UserManager<User> userManager;
+        private readonly OutOfSchoolDbContext context;
+        private readonly IRazorViewToStringRenderer renderer;
+
+        public MinistryAdminService(
+            IMapper mapper,
+            IMinistryAdminRepository ministryAdminRepository,
+            ILogger<MinistryAdminService> logger,
+            IEmailSender emailSender,
+            UserManager<User> userManager,
+            OutOfSchoolDbContext context,
+            IRazorViewToStringRenderer renderer)
+        {
+            this.mapper = mapper;
+            this.userManager = userManager;
+            this.context = context;
+            this.ministryAdminRepository = ministryAdminRepository;
+            this.logger = logger;
+            this.emailSender = emailSender;
+            this.renderer = renderer;
+        }
+
+        public async Task<ResponseDto> CreateMinistryAdminAsync(
+            CreateMinistryAdminDto ministryAdminDto,
```

Comment:
I added ModelState check in Controller so it's not need to check every field in in service.

Reply:
That's wrong that you're trying to bind your service with your particular method in controller. Your controller and the service are different classes. Moreover they are separated conceptually, I mean these concrete classes stem from logically separated layers (Presentation and Business Logic Layers) even if we hold them in one project.
Here I would like to say next things:
1) What if I want to add some method into your controller let me say in half a year. Of course most likely I'll check ModelState in my method but if my case allows nulls (some reasons) I can be confused when our app suddenly will throw `NullReferenceException`  that is much more incomprehensible than `ArgumentNullException` describing the reason.
2) Now more realistic case: if I wanna use your service for another purpose (in another service). Should I jump into your source code and check everywhere that you handle parameters you don't want to get? I think it's strange because if I pass some parameters that are wrong I expect either a result or an exception but understandable one. The reason that `NullReferenceException` sometimes say nothing about what happenned and you need to complete rigorous debug.

Reply:
Thank you for detailed explanation.
Fixed in new PR: https://github.com/ita-social-projects/OoS-Backend/pull/747